narwhals/_arrow/dataframe.py

```diff
@@ -383,11 +383,10 @@ def join_asof(
         *,
         left_on: str | None,
         right_on: str | None,
-        on: str | None,
-        by_left: str | list[str] | None,
-        by_right: str | list[str] | None,
-        by: str | list[str] | None,
+        by_left: list[str] | None,
+        by_right: list[str] | None,
         strategy: Literal["backward", "forward", "nearest"],
+        suffix: str = "_right",
```

Comment:
really minor, but can we avoid setting the default in the compliant level? So, in `narwhals.dataframe.py`, we can have `suffix: str = "_right"`, but in the internal methods in `_arrow` / `_duckdb` / `_pandas_like` / etc., we can just have `suffix: str`, to make sure that we're always passing it down?